apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/grpc/StringifiedValueUtil.java

```diff
@@ -172,6 +172,132 @@ public static void decodeStringifiedMap(
         String.format("Invalid map value '%s': missing closing '}'", value));
   }
 
+  public static void decodeStringifiedMap(
+      String value,
+      ToProtoValueCodec keyCodec,
+      ToProtoValueCodec valueCodec,
+      Collection<QueryOuterClass.Value> results) {
+    int idx = skipSpaces(value, 0);
+    if (idx >= value.length()) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Invalid Map value '%s': at character %d expecting '[' but got EOF", value, idx));
```

Comment:
Could the exception message indicate that this is "non-compact" map, so when we see decode problems it's clear which mode value was being decoded in? Similarly for existing code, indicating "compact" map representation being decoded.